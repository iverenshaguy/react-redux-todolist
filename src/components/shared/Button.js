import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  return (
    <button
      type={props.buttonType || 'button'}
      className={props.btnClass || 'btn btn-primary'}
      onClick={() => props.onClick()}
    >
      {props.buttonText}
    </button>
  );
};

Button.proptypes = {
  buttonType: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func
};

export default Button;
