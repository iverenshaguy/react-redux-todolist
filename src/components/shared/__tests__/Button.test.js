import Button from '../Button';

const clickFunction = jest.fn();
const setup = () => {
  const props = {
    onClick: clickFunction,
    type: 'button',
    className: 'btnClassName'
  };
  const shallowWrapper = shallow(<Button {...props} />);
  const mountedWrapper = mount(<Button {...props} />);

  return {
    shallowWrapper,
    mountedWrapper,
    props
  };
};

describe('Button', () => {
  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const { mountedWrapper } = setup();
    expect(toJson(mountedWrapper)).toMatchSnapshot();
  });

  it('calls a function', () => {
    const { mountedWrapper } = setup();
    mountedWrapper.simulate('click');
    expect(clickFunction).toHaveBeenCalled();
  });
});
