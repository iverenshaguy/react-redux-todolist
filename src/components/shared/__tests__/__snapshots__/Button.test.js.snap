// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button mounts correctly 1`] = `
<Button
  className="btnClassName"
  onClick={[Function]}
  type="button"
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  />
</Button>
`;

exports[`Button renders correctly 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
  type="button"
/>
`;
